cirq/circuits/circuit.py

```diff
@@ -787,16 +787,17 @@ def _can_commute_past(self,
     def insert(
             self,
             index: int,
-            moment_or_operation_tree: Union[Moment, ops.OP_TREE],
+            moment_or_operation_tree: ops.OP_TREE,
```

Comment:
This is not the correct type.

I recommend defining `OP_OR_MOMENT_TREE = Union[Operation, Moment, OP_TREE, Iterable[Any]]` in order to appease mypy.

Reply:
Why isn't it the correct type? Operation and Moment are both examples of OP_TREE.

Reply:
That's... Huh. I guess that's right. It just seems really weird to tell the user you're asking for "operations or moments" and then just say op tree.

Reply:
That's why I initially renamed to variable to just `operations`. Also, mypy was mainly unhappy about the return type of `flatten_op_tree` being changed from `Iterable[Operation]` to `Iterable[Union[Operations, Moment]]`.

Reply:
You would need to define overloads that said OP_TREE mapped to OP_TREE while the more complicated type mapped to the more complicated type.

Reply:
@Strilanc do you still want me to define `OP_OR_MOMENT_TREE` or are you okay with just leaving `OP_TREE` and changing the variable name to simply `operations`?

Reply:
In my opinion the variable name needs to say moments because that's important documentation that will affect user expectations. And the type name should probably also say moments. So yes, please change it.

Reply:
I've renamed the variable to `moment_or_operation_tree` and the type to `Union[Moment, OP_TREE]`.